core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -162,6 +162,7 @@ class Partition(val topic: String,
    */
   def makeLeader(controllerId: Int,
                  partitionStateInfo: PartitionStateInfo, correlationId: Int): Boolean = {
+    var leaderHWIncremented = false
```

Comment:
`inWriteLock` returns a value, so you can simply use a val instead of a `var` here.